ipa-core/src/report/hybrid.rs

```diff
@@ -308,6 +441,282 @@ where
+
+        Ok(HybridConversionReport::<V> {
+            match_key: Replicated::<BA64>::deserialize_infallible(GenericArray::from_slice(
+                plaintext_mk,
+            )),
+            value: Replicated::<V>::deserialize(GenericArray::from_slice(plaintext_btt)).map_err(
+                |e| InvalidHybridReportError::DeserializationError("trigger_value", e.into()),
+            )?,
+        })
+    }
+}
+
+#[derive(Clone, Eq, PartialEq)]
+pub enum EncryptedHybridGeneralReport<BK, V>
+where
+    BK: SharedValue,
+    V: SharedValue,
+{
+    Impression(EncryptedHybridImpressionReport<BK>),
+    Conversion(EncryptedHybridConversionReport<V>),
+}
+
+impl<BK, V> EncryptedHybridGeneralReport<BK, V>
+where
+    V: SharedValue,
+    BK: SharedValue,
+    Replicated<V>: Serializable,
+    Replicated<BK>: Serializable,
+    <Replicated<V> as Serializable>::Size: Add<U16>,
+    <<Replicated<V> as Serializable>::Size as Add<U16>>::Output: ArrayLength,
+    <Replicated<BK> as Serializable>::Size: Add<U16>,
+    <<Replicated<BK> as Serializable>::Size as Add<U16>>::Output: ArrayLength,
+{
+    pub fn encap_key_mk(&self) -> &[u8] {
+        match self {
+            EncryptedHybridGeneralReport::Impression(impression_report) => {
+                impression_report.encap_key_mk()
+            }
+            EncryptedHybridGeneralReport::Conversion(conversion_report) => {
+                conversion_report.encap_key_mk()
+            }
+        }
+    }
+
+    pub fn mk_ciphertext(&self) -> &[u8] {
+        match self {
+            EncryptedHybridGeneralReport::Impression(impression_report) => {
+                impression_report.mk_ciphertext()
+            }
+            EncryptedHybridGeneralReport::Conversion(conversion_report) => {
+                conversion_report.mk_ciphertext()
+            }
+        }
+    }
+
+    pub fn encap_key_btt(&self) -> &[u8] {
+        match self {
+            EncryptedHybridGeneralReport::Impression(impression_report) => {
+                impression_report.encap_key_btt()
+            }
+            EncryptedHybridGeneralReport::Conversion(conversion_report) => {
+                conversion_report.encap_key_btt()
+            }
+        }
+    }
+
+    pub fn btt_ciphertext(&self) -> &[u8] {
+        match self {
+            EncryptedHybridGeneralReport::Impression(impression_report) => {
+                impression_report.btt_ciphertext()
+            }
+            EncryptedHybridGeneralReport::Conversion(conversion_report) => {
+                conversion_report.btt_ciphertext()
+            }
+        }
+    }
+
+    pub fn key_id(&self) -> u8 {
+        match self {
+            EncryptedHybridGeneralReport::Impression(impression_report) => {
+                impression_report.key_id()
+            }
+            EncryptedHybridGeneralReport::Conversion(conversion_report) => {
+                conversion_report.key_id()
+            }
+        }
+    }
```

Comment:
do we actually need access to these on the `EncryptedHybridGeneralReport`? I think we should only need `decrypt` and `from_bytes` (which call into the enum items).

Reply:
I saw that mk_ciphertext was needed for UniqueBytes. Not sure about the other ones though (or if we might need other ones later?)